core/controllers/acl_decorators.py

```diff
@@ -2933,3 +2933,82 @@ def test_request_originates_from_valid_oppia_ml_instance(self, **kwargs):
     test_request_originates_from_valid_oppia_ml_instance.__wrapped__ = True
 
     return test_request_originates_from_valid_oppia_ml_instance
+
+
+def can_update_suggestions(handler):
+    """Decorator to check whether the current user can update suggestions.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that has common checks and
+        permissions specified by passed in decorator. This function is allowed
+        to be called for users who have the rights to accept suggestions and
+        the suggesters who made their own suggestions.
+
+    Raises:
+        NotLoggedInException. The user is not logged in.
+        UnauthorizedUserException. The user does not have credentials to
+            edit this suggestion.
+        InvalidInputException. The submitted suggestion id is not valid.
+        PageNotFoundException. A suggestion is not found with the given
+            suggestion id.
+    """
+    def test_can_update_suggestion(
+            self, suggestion_id, **kwargs):
+        """Returns a handler to test whether a
+        suggestion can be updated based on the user's roles.
+
+        Args:
+            suggestion_id: str. The suggestion id.
+            **kwargs: *. Keyword arguments.
+
+        Returns:
+            function. The handler for updating a suggestion.
+
+        Raises:
+            NotLoggedInException. The user is not logged in.
+            UnauthorizedUserException. The user does not have credentials to
+                edit this suggestion.
+            InvalidInputException. The submitted suggestion id is not valid.
+            PageNotFoundException. A suggestion is not found with the given
+                suggestion id.
+        """
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+        user_actions = self.user.actions
+
+        if len(suggestion_id.split('.')) != 3:
+            raise self.InvalidInputException(
+                'Invalid format for suggestion_id.'
+                ' It must contain 3 parts separated by \'.\'')
+
+        suggestion = suggestion_services.get_suggestion_by_id(suggestion_id)
+
+        if suggestion is None:
+            raise self.PageNotFoundException
+
+        # TODO(#11735): Currently, only checks whether users can review
+        # translation sugeestions. Another check should be added to check
+        # whether users can review question suggestions if the suggestion
+        # type is add question.
```

Comment:
Added a TODO comment regarding the original issue and the need of generalizing this for questions.
Refernce: https://github.com/oppia/oppia/pull/12424/files#r612952710
/cc @seanlip

Reply:
Shouldn't this be removed or a different issue referenced? This PR is indicating #11735 is being fixed, so there shouldn't be anything left to do. Or, should the PR be updated to indicate it's only fixing part of #11735?

Reply:
@chris7716 Given that we aren't rushing this into the April release any more, can this PR be edited to be generalizable for questions as well (at least for this decorator)? I don't see the point of introducing an interim state here. /cc @sagangwee

Reply:
@seanlip That sounds good to me. Talked to Hasitha offline, we'll aim to update this PR with the corresponding question editing backend changes by next week, conservatively leaving one week for review by the May release cut (05-07). The frontend changes for editing question suggestions is out of scope for this PR though.